eng/tools/typespec-validation/src/tspconfig-rule-helper.ts

```diff
@@ -0,0 +1,154 @@
+import { Rule } from "./rule.js";
+import { RuleResult } from "./rule-result.js";
+import { TsvHost } from "./tsv-host.js";
+import { join } from "path";
+import { parse as yamlParse } from "yaml";
+
+type ExpectedValueType = string | boolean | RegExp;
+
+class TspconfigRuleBase implements Rule {
+  name: string;
+  description: string;
+
+  protected keyToValidate: string;
+  protected expectedValue: ExpectedValueType;
+
+  constructor(
+    ruleName: string,
+    description: string,
+    keyToValidate: string,
+    expectedValue: ExpectedValueType,
+  ) {
+    this.name = ruleName;
+    this.description = description;
+    this.keyToValidate = keyToValidate;
+    this.expectedValue = expectedValue;
+  }
+
+  public async execute(host: TsvHost, folder: string): Promise<RuleResult> {
+    const tspconfigExists = await host.checkFileExists(join(folder, "tspconfig.yaml"));
+    if (!tspconfigExists)
+      return this.createFailedResult(
+        `Failed to find ${join(folder, "tspconfig.yaml")}`,
+        "Please add tspconfig.yaml",
+      );
+
+    let config = undefined;
+    try {
+      const configText = await host.readTspConfig(folder);
+      config = yamlParse(configText);
+    } catch (error) {
+      return this.createFailedResult(
+        `Failed to parse ${join(folder, "tspconfig.yaml")}`,
+        "Please add tspconfig.yaml.",
+      );
+    }
+
+    const { shouldSkip, skipResult } = this.skip(config);
+    if (shouldSkip) return skipResult;
+    return this.validate(config);
+  }
+
+  protected skip(_: any): { shouldSkip: boolean; skipResult: RuleResult } {
```

Comment:
What is the purpose of this method?  It ignores it's parameter and always returns false.

Reply:
e.g. some rules to be add will only apply to management plane. It's prepared for these cases